### PR TITLE
FIX - feat(azure): Add Azure Scheduled Commands support

### PR DESCRIPTION
**Pull request by gonzalogarciajaubert**
## Note
This PR fixes the previous one https://github.com/boostercloud/booster/pull/965 that was reverted on and fixed here

## Description
There’s a Scheduled Commands feature in AWS that takes a typical cron string and triggers a command on a schedule instead of making it wait for a request. This feature is now supported in Azure

## Changes

* Add PackageAzureFunction classes for: GraphQL, EventHandler and ScheduleCommands function
* Refactor Azure/ApplicationStackBuilder class
* Refactor Azure/FunctionDefinition
* Add Azure rawScheduledInputToEnvelope
* Update Azure Provider to include Azure/rawScheduledInputToEnvelope call

## Checks
- [X] Project Builds
- [X] Project passes tests and checks
- [X] Updated documentation accordingly

### CODE REVIEW

**Review by rdiaz82 — APPROVED** [review]
LGTM

File: packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/application-stack.ts [hunk 1]
```diff
@@ -92,48 +95,14 @@ export class ApplicationStackBuilder {
...
-      },
-    ])
+    const graphqlFunctionDefinition = new GraphqlFunction(this.config).getFunctionDefinition()
+    const eventHandlerFunctionDefinition = new EventHandlerFunction(this.config).getFunctionDefinition()
+    let featuresDefinitions = [graphqlFunctionDefinition, eventHandlerFunctionDefinition]
+    const schedulesFunctionsDefinition = new SchedulesFunctions(this.config).getFunctionDefinitions()
+    if (schedulesFunctionsDefinition) {
+      featuresDefinitions = featuresDefinitions.concat(schedulesFunctionsDefinition)
+    }
+    const zipPath = await packageAzureFunction(featuresDefinitions)
```
Comment (alvaroloes) [inline packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/application-stack.ts:105]: good refactor!

File: packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/event-handler-function.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            name: 'rawEvent',
+            direction: 'in',
+            leaseCollectionName: 'leases',
+            connectionStringSetting: 'COSMOSDB_CONNECTION_STRING',
+            databaseName: this.config.resourceNames.applicationStack,
+            collectionName: this.config.resourceNames.eventsStore,
+            createLeaseCollectionIfNotExists: 'true',
+          },
+        ],
+        scriptFile: '../dist/index.js',
```
Comment (alvaroloes) [inline packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/event-handler-function.ts:23]: I would say using `this.config.userProjectRootPath` would be safer. Would it be possible to use it here? (and in the other places where this string is used)

File: packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/schedules-functions.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      config: {
+        bindings: [
+          {
+            type: 'timerTrigger',
+            name: `${scheduledCommandInfo.name}`,
+            direction: 'in',
+            schedule: `${cronExpression}`,
+          },
+        ],
+        scriptFile: '../dist/index.js',
```
Comment (alvaroloes) [inline packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/schedules-functions.ts:37]: Same comment here (and in other places too)

File: packages/framework-provider-azure/src/index.ts [hunk 4]
```diff
@@ -67,7 +68,7 @@ export const Provider = (): ProviderLibrary => ({
   },
   // ScheduledCommandsLibrary
   scheduled: {
-    rawToEnvelope: undefined as any,
+    rawToEnvelope: rawScheduledInputToEnvelope,
```
Comment (alvaroloes) [inline packages/framework-provider-azure/src/index.ts:71]: 🎉

**Review by alvaroloes — APPROVED** [review]
What a great work!
Approved, but please consider the suggestion about `this.config.userProjectRootPath`

File: packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/event-handler-function.ts (hunk 2 above)
Comment (gonzalogarciajaubert) [reply packages/framework-provider-azure-infrastructure/src/infrastructure/stacks/event-handler-function.ts:23]: Correct me if I'm wrong but this property returns the full project path. Example: `/Users/gonzalogarcia/dev/src/theam/booster_projects/booster-az-cart-demo/.deploy/`. Am I using it wrongly?

**Pull request merged**